pkg/receive/multitsdb.go

```diff
@@ -198,6 +199,84 @@ func (t *MultiTSDB) Close() error {
 	return merr.Err()
 }
 
+// Prune flushes and closes the TSDB for tenants that haven't received
+// any new samples for longer than the TSDB retention period.
+func (t *MultiTSDB) Prune(ctx context.Context) error {
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+
+	var (
+		wg   sync.WaitGroup
+		merr errutil.SyncMultiError
```

Comment:
`SyncMultiError` is really neat! I wonder if we should use it for [`Flush()`](https://github.com/thanos-io/thanos/blob/main/pkg/receive/multitsdb.go#L172) and [`Sync()`](https://github.com/thanos-io/thanos/blob/main/pkg/receive/multitsdb.go#L226) in multitsdb.go too?

Reply:
We could do it, I was just afraid to not introduce a bug or a race condition by refactoring something out of scope for this feature :)

Reply:
👍🏽